third_party/move/extensions/move-table-extension/src/lib.rs

```diff
@@ -88,11 +88,28 @@ pub struct TableChange {
 /// A table resolver which needs to be provided by the environment. This allows to lookup
 /// data in remote storage, as well as retrieve cost of table operations.
 pub trait TableResolver {
-    fn resolve_table_entry(
+    // Note:
+    //
+    // This methods takes an optional layout argument to handle aggregator liftings.
+    // If it is None:
+    //   - The method simply returns bytes of a table item.
+    // If it is Some(..):
+    //   - Every location which is tagged as aggregator lifting is replaced with unique identifiers.
+    //   - The returned table item bytes do not have real values at these locations anymore.
+    fn resolve_table_entry_with_layout(
         &self,
         handle: &TableHandle,
         key: &[u8],
+        layout: Option<&MoveTypeLayout>,
```

Comment:
Same question here (but out of order lol), what's the trade-off with passing non-option layout and requiring the caller to call the other method otherwise?

Reply:
Did that option